File: src/lessons/welcome/loop/dowhileloop/LoopDoWhileEntity.java
package lessons.welcome.loop.dowhileloop;

import java.awt.Color;

public class LoopDoWhileEntity extends jlm.universe.bugglequest.SimpleBuggle {
	boolean isGroundWhite() { 
		return getGroundColor().equals(Color.white)?true:false;
	}
	@Override
	/* BEGIN TEMPLATE */
	public void run() { 
		/* BEGIN SOLUTION */
		do {
			forward();
		} while (!isGroundWhite());
		/* END SOLUTION */
	}
	/* END TEMPLATE */
}

File: src/lessons/welcome/loopdowhile/LoopDoWhileEntity.scala
package lessons.welcome.loopdowhile;

import java.awt.Color;

class LoopDoWhileEntity extends plm.universe.bugglequest.SimpleBuggle {
	/* BINDINGS TRANSLATION */
	def estSurBlanc():Boolean = { return isGroundWhite(); }

	override def run() {
		/* BEGIN SOLUTION */
		do {
			stepForward();
		} while (!isGroundWhite());
		/* END SOLUTION */
	}
}
